From 08da1edf95473a54c665b44f62a98fa2520a681f Mon Sep 17 00:00:00 2001
From: Chen Yu <yu.c.chen@intel.com>
Date: Wed, 19 Feb 2025 09:25:02 +0200
Subject: [PATCH 1/2] hack for 2 SNC

Signed-off-by: Chen Yu <yu.c.chen@intel.com>
---
 .../src/autonumabench/autonumabench-install    | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/shellpack_src/src/autonumabench/autonumabench-install b/shellpack_src/src/autonumabench/autonumabench-install
index fec79941..06611279 100755
--- a/shellpack_src/src/autonumabench/autonumabench-install
+++ b/shellpack_src/src/autonumabench/autonumabench-install
@@ -2,7 +2,7 @@
 # autonumabench installer
 
 ###SHELLPACK preamble autonumabench-install 0
-GIT_LOCATION=https://gitorious.org/autonuma-benchmark/autonuma-benchmark.git
+GIT_LOCATION=https://github.com/pholasek/autonuma-benchmark.git
 MIRROR_LOCATION="$WEBROOT/autonumabench/"
 
 ###SHELLPACK parseargBegin
@@ -99,6 +99,20 @@ index 101286e..5780b09 100755
 +			NUMA01_TALLOC=1
  			HARDBIND=1
  			INVERSEBIND=1
- 			;;' | patch -p1 || die Failed to patch autonumabench
+ 			;;
+diff --git a/preproc.awk b/preproc.awk
+index 41c7e60..d6bc498 100644
+--- a/preproc.awk
++++ b/preproc.awk
+@@ -91,7 +91,7 @@ END \
+ 	cpus += 1
+ 	if (MoF)
+ 		cpus = MoF
+-	threads = "#define THREADS "(cpus / 2)
++	threads = "#define THREADS "(cpus / 4)
+ 	ncpus = "#define NCPUS "cpus
+ 	nnodes = "#define NNODES "(nodes)
+ 	if (file == "numa01.c")
+			' | patch -p1 || die Failed to patch autonumabench
 
 echo autonumabench installed successfully
-- 
2.48.1

