diff --git a/linux-headers/linux/kvm.h b/linux-headers/linux/kvm.h
index 9bd384e1fa..68c59da310 100644
--- a/linux-headers/linux/kvm.h
+++ b/linux-headers/linux/kvm.h
@@ -15,7 +15,7 @@
 #include <asm/kvm.h>
 
 #define KVM_API_VERSION 12
-
+#define KVM_GUEST_MEMFD_ALLOW_HUGEPAGE		(1ULL << 0)
 /* *** Deprecated interfaces *** */
 
 #define KVM_TRC_SHIFT           16
diff --git a/system/physmem.c b/system/physmem.c
index 4af2c7ddbb..b97a0bd27c 100644
--- a/system/physmem.c
+++ b/system/physmem.c
@@ -1876,8 +1876,9 @@ static void ram_block_add(RAMBlock *new_block, Error **errp)
     if (kvm_enabled() && (new_block->flags & RAM_GUEST_MEMFD) &&
         new_block->guest_memfd < 0) {
         /* TODO: to decide if KVM_GUEST_MEMFD_ALLOW_HUGEPAGE is supported */
+	uint64_t flags = KVM_GUEST_MEMFD_ALLOW_HUGEPAGE;
         new_block->guest_memfd = kvm_create_guest_memfd(new_block->max_length,
-                                                        0, errp);
+                                                        flags, errp);
         if (new_block->guest_memfd < 0) {
             qemu_mutex_unlock_ramlist();
             return;
-- 
2.34.1

